File: src/main/assets/Assets.java
package main.assets;

import com.raylib.Raylib;
import com.raylib.Raylib.Music;
import com.raylib.Raylib.Sound;
import com.raylib.Raylib.Texture;


public class Assets {
	
	public static Texture player;

	public static Texture raft0;
	public static Texture raft1;
	public static Texture raft2;
	public static Texture water;

	//? X for Xbox
	//? P for PlayStation
	public static Texture controllerLeftX;
	public static Texture controllerLeftP;

	public static Texture controllerRightX;
	public static Texture controllerRightP;

	public static Texture controllerUpX;
	public static Texture controllerUpP;

	public static Texture controllerDownX;
	public static Texture controllerDownP;

	public static Music music;

	// Load all the assets
	public static void loadAssets() {

		// Get the base path to the textures
		String textures = "./res/texture/";

		player = Raylib.LoadTexture(textures + "player.png");

		raft0 = Raylib.LoadTexture(textures + "raft0.png");
		raft1 = Raylib.LoadTexture(textures + "raft1.png");
		raft2 = Raylib.LoadTexture(textures + "raft2.png");
		water = Raylib.LoadTexture(textures + "water.png");

		// Controller textures
		String controllerTextures = "./res/texture/controller/";
		
		controllerLeftX = Raylib.LoadTexture(controllerTextures + "x_xbox.png");
		controllerRightX = Raylib.LoadTexture(controllerTextures + "b_xbox.png");
		controllerUpX = Raylib.LoadTexture(controllerTextures + "y_xbox.png");
		controllerDownX = Raylib.LoadTexture(controllerTextures + "a_xbox.png");

		controllerLeftP = Raylib.LoadTexture(controllerTextures + "square_ps.png");
		controllerRightP = Raylib.LoadTexture(controllerTextures + "circle_ps.png");
		controllerUpP = Raylib.LoadTexture(controllerTextures + "triangle_ps.png");
		controllerDownP = Raylib.LoadTexture(controllerTextures + "x_ps.png");

		// Get the base path to the audio
		String sound = "./res/sound/";
		
		music = Raylib.LoadMusicStream(sound + "music.wav");

	}

	// Unload all the assets
	public static void unloadAssets() {

		Raylib.UnloadTexture(player);
		Raylib.UnloadTexture(raft0);
		Raylib.UnloadTexture(raft1);
		Raylib.UnloadTexture(raft2);
		Raylib.UnloadTexture(water);
		Raylib.UnloadTexture(controllerLeftX);
		Raylib.UnloadTexture(controllerLeftP);
		Raylib.UnloadTexture(controllerRightX);
		Raylib.UnloadTexture(controllerRightP);
		Raylib.UnloadTexture(controllerUpX);
		Raylib.UnloadTexture(controllerUpP);
		Raylib.UnloadTexture(controllerDownX);
		Raylib.UnloadTexture(controllerDownP);
		Raylib.UnloadMusicStream(music);
	}
}

File: src/assets/Assets.cs
using Raylib_cs;

class Assets
{
	public static Texture2D Player1 { get; private set; }
	public static Texture2D Player2 { get; private set; }
	public static Texture2D Player3 { get; private set; }
	public static Texture2D Player4 { get; private set; }

	public static Texture2D LobbyBackground { get; private set; }
	public static Texture2D Raft0 { get; private set; }
	public static Texture2D Raft1 { get; private set; }
	public static Texture2D Raft2 { get; private set; }
	public static Texture2D Water { get; private set; }

	//? x for xbox
	public static Texture2D ControllerLeftX { get; private set; }
	public static Texture2D ControllerRightX { get; private set; }
	public static Texture2D ControllerUpX { get; private set; }
	public static Texture2D ControllerDownX { get; private set; }

	//? p for playstation
	public static Texture2D ControllerLeftP { get; private set; }
	public static Texture2D ControllerRightP { get; private set; }
	public static Texture2D ControllerUpP { get; private set; }
	public static Texture2D ControllerDownP { get; private set; }

	// TODO: Put this in AssetManager class, and just have
	// TODO: this for keeping the variables. Bit weird rn using
	// TODO: Assets.LoadAssets and AssetManager.UnloadAssets
	public static void LoadAssets()
	{
		// Player textures
		Player1 = AssetManager.LoadTexture(AssetManager.TexturePath, "player1");
		Player2 = AssetManager.LoadTexture(AssetManager.TexturePath, "player2");
		Player3 = AssetManager.LoadTexture(AssetManager.TexturePath, "player3");
		Player4 = AssetManager.LoadTexture(AssetManager.TexturePath, "player4");

		// Environmental textures
		LobbyBackground = AssetManager.LoadTexture(AssetManager.TexturePath, "background");
		Raft0 = AssetManager.LoadTexture(AssetManager.TexturePath, "raft0");
		Raft1 = AssetManager.LoadTexture(AssetManager.TexturePath, "raft1");
		Raft2 = AssetManager.LoadTexture(AssetManager.TexturePath, "raft2");
		Water = AssetManager.LoadTexture(AssetManager.TexturePath, "water");

		// Xbox controller textures
		ControllerLeftX = AssetManager.LoadTexture(AssetManager.ControllerTexturePath, "left_x");
		ControllerRightX = AssetManager.LoadTexture(AssetManager.ControllerTexturePath, "right_x");
		ControllerUpX = AssetManager.LoadTexture(AssetManager.ControllerTexturePath, "up_x");
		ControllerDownX = AssetManager.LoadTexture(AssetManager.ControllerTexturePath, "down_x");

		// PlayStation controller textures
		ControllerLeftP = AssetManager.LoadTexture(AssetManager.ControllerTexturePath, "left_p");
		ControllerRightP = AssetManager.LoadTexture(AssetManager.ControllerTexturePath, "right_p");
		ControllerUpP = AssetManager.LoadTexture(AssetManager.ControllerTexturePath, "up_p");
		ControllerDownP = AssetManager.LoadTexture(AssetManager.ControllerTexturePath, "down_p");
	}
}